lib/updateDeps.js

```diff
@@ -0,0 +1,166 @@
+ *
+ * @param {Package} pkg Package object.
+ * @param {string|undefined} bumpStrategy Dependency resolution strategy: override, satisfy, inherit.
+ * @param {string|undefined} releaseStrategy Release type triggered by deps updating: patch, minor, major, inherit.
+ * @param {Package[]} ignore=[] Packages to ignore (to prevent infinite loops).
+ * @returns {string|undefined} Resolved release type.
+ * @internal
+ */
+const resolveReleaseType = (pkg, bumpStrategy = "override", releaseStrategy = "patch", ignore = []) => {
+	// NOTE This fn also updates pkg deps, so it must be invoked anyway.
+	const dependentReleaseType = getDependentRelease(pkg, bumpStrategy, releaseStrategy, ignore);
+
+	// Release type found by commitAnalyzer.
+	if (pkg._nextType) {
+		return pkg._nextType;
+	}
+
+	if (!dependentReleaseType) {
+		return undefined;
+	}
+
+	pkg._nextType = releaseStrategy === "inherit" ? dependentReleaseType : releaseStrategy;
+
+	return pkg._nextType;
+};
+
+/**
+ * Get dependent release type by recursive scanning and updating its deps.
+ *
+ * @param {Package} pkg The package with local deps to check.
+ * @param {string} bumpStrategy Dependency resolution strategy: override, satisfy, inherit.
+ * @param {string} releaseStrategy Release type triggered by deps updating: patch, minor, major, inherit.
+ * @param {Package[]} ignore Packages to ignore (to prevent infinite loops).
+ * @returns {string|undefined} Returns the highest release type if found, undefined otherwise
+ * @internal
+ */
+const getDependentRelease = (pkg, bumpStrategy, releaseStrategy, ignore) => {
+	const severityOrder = ["patch", "minor", "major"];
+	const { localDeps, manifest = {} } = pkg;
+	const { dependencies = {}, devDependencies = {}, peerDependencies = {}, optionalDependencies = {} } = manifest;
+	const scopes = [dependencies, devDependencies, peerDependencies, optionalDependencies];
+	const bumpDependency = (scope, name, nextVersion) => {
+		const currentVersion = scope[name];
+		if (!nextVersion || !currentVersion) {
+			return;
+		}
+		const resolvedVersion = resolveNextVersion(currentVersion, nextVersion, releaseStrategy);
+
+		if (currentVersion !== resolvedVersion) {
+			scope[name] = resolvedVersion;
+
+			return true;
+		}
+	};
+
+	// prettier-ignore
+	return localDeps
+		.filter((p) => ignore.indexOf(p) === -1)
+		.reduce((releaseType, p) => {
+			const name = p.name;
+
+			// Has changed if...
+			// 1. Any local dep package itself has changed
+			// 2. Any local dep package has local deps that have changed.
+			const nextType = resolveReleaseType(p, bumpStrategy, releaseStrategy,[...ignore, ...localDeps]);
+			const nextVersion = getNextVersion(p);
+			const lastVersion = pkg._lastRelease && pkg._lastRelease.version;
+
+			// 3. And this change should correspond to manifest updating rule.
+			const requireRelease = [
+				...scopes.map((scope) => bumpDependency(scope, name, nextVersion)),
```

Comment:
If you make `requireRelease` into a variable passed in, then you could mutate it, meaning you'd not need to do the spread and some check; The returns in bumpDependency would just become assignments, I think — might need more restructuring though.

do you need to update all scopes if there'll be a release, or can you just do `scopes.some(`?

Reply:
Ah. Spread is completely useless here. https://github.com/dhoulb/multi-semantic-release/pull/45/commits/d50a9e47d549113b6b3b0f6db109e451884df3a5

I think all scopes should be updated.